lib/src/util/cast_ui_factory.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:over_react/component_base.dart';
+
+/// Casts [value] to `UiFactory<T>`, which can be inferred from the surrounding context,
+/// allowing you to perform casts of dynamic values without having to repeat the type.
+///
+/// Example:
+///
+/// ```dart
+/// UiFactory<FooProps> Foo = castUiFactory(_$Foo);
+/// ```
+///
+/// This is necessary because ungenerated members (identifiable by the `_$` prefix)
+/// are considered dynamic. This requires a cast on the right hand side. However,
+/// due to complications around formatting and "ignore" comments, in order to prevent
+/// false positive lints or errors, there are limitations around using an explicit
+/// cast (i.e. `_$Foo as UiFactory<FooProps>`).
```

Comment:
In the first part, it's unclear what "This" is referring to. Also, the last sentence is a little hard to follow, and doesn't flow well (see note above about moving it up).

How about something like:

> Casting the generated factory is necessary because, until code generation is run, the generated factory is of type is `dynamic`, meaning that the following code would emit an implicit_cast error (which can't be ignored as of Dart 2.9):
> ```
> UiFactory<FooProps> Foo = _$Foo; // ignore: undefined_identifier
> ```